src/main/java/gift/MemberController.java

```diff
@@ -0,0 +1,48 @@
+package gift;
+
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/member")
+public class MemberController {
+
+    private final MemberDao memberDao;
+    private final JwtTokenUtil jwtTokenUtil;
+
+    public MemberController(MemberDao memberDao, JwtTokenUtil jwtTokenUtil) {
+        this.memberDao = memberDao;
+        this.jwtTokenUtil = jwtTokenUtil;
+    }
+
+    @PostMapping("/register")
+    public ResponseEntity<?> registerMember(@RequestBody Member member) {
+        Optional<Member> existMember = memberDao.findMember(member);
+        if (!existMember.isPresent()) {
+            memberDao.insertMember(member);
+            String token = jwtTokenUtil.generateToken(member.getEmail());
```

Comment:
```suggestion
            String token = JwtTokenUtil.generateToken(member.getEmail());
```

`JwtTokenUtil` 클래스는 이름에서 나타내주셨듯  utility 클래스이기 때문에 bean 으로 관리할 필요도 없고,
객체를 만들어서 사용하지 않아도 돼요.
그러니 빈으로 등록하지 말고 사용해보면 어떨까요? 제가 제시한 부분 말고 다른 부분도 약간 변경해야할 것 같네요!

* 빈으로 등록하지 않기
* `MemberController` 에 주입해주는 부분을 제거하기

이 두가지를 해보시면 됩니다!

Reply:
멘토님 덕분에 유틸리티 클래스에 대해서 다시 한번 이해하고 넘어갈 수 있었습니다:)구체적인 제시 감사합니다!

Reply:
반영했습니다!